src/plugins/data_source_management/public/components/direct_query_data_sources_components/associated_object_management/utils/associated_objects_tab_utils.tsx

```diff
@@ -43,49 +43,38 @@ export const isCatalogCacheFetching = (...statuses: DirectQueryLoadingStatus[])
   );
 };
 
-export const redirectToExplorerWithDataSrc = (
+export const redirectToDiscoverWithDataSrc = (
   datasourceName: string,
-  datasourceType: string,
+  datasourceMDSId: string | undefined,
   databaseName: string,
   tableName: string,
   application: ApplicationStart
 ) => {
-  const queryIndex = `${datasourceName}.${databaseName}.${tableName}`;
-  redirectToExplorerWithQuery(datasourceName, datasourceType, queryIndex, application);
-};
-
-export const redirectToExplorerOSIdx = (indexName: string, application: ApplicationStart) => {
-  redirectToExplorerWithQuery(
-    DEFAULT_DATA_SOURCE_NAME,
-    DEFAULT_DATA_SOURCE_TYPE,
-    indexName,
-    application
-  );
-};
-
-export const redirectToExplorerS3 = (datasourceName: string, application: ApplicationStart) => {
-  application.navigateToApp(observabilityLogsID, {
-    path: `#/explorer`,
-    state: {
-      datasourceName,
-      datasourceType: DATA_SOURCE_TYPES.S3Glue,
-    },
+  application.navigateToApp('data-explorer', {
+    path: `discover#?_a=(discover:(columns:!(_source),isDirty:!f,sort:!()),metadata:(view:discover))&_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now-15m,to:now))&_q=(filters:!(),query:(dataset:(dataSource:(id:'${
+      datasourceMDSId ?? ''
+    }',meta:(name:${datasourceName},type:CUSTOM),title:'',type:DATA_SOURCE),id:'${
+      datasourceMDSId ?? ''
+    }::${datasourceName}.${databaseName}.${tableName}',title:${datasourceName}.${databaseName}.${tableName},type:S3),language:SQL,query:'SELECT%20*%20FROM%20${datasourceName}.${databaseName}.${tableName}%20LIMIT%2010'))`,
```

Comment:
Does this re-direction also needs the `query:enhancements:enabled` to be `true` to work?

Reply:
It would need it to be true since the SQL language would be hidden under that setting

Reply:
Got it. So what happened if we disabled the query enhancement and perform a redirection in data source management?

Reply:
I've thought about it and I think the best solution here is to stop redirection with maybe a toast that says to enable query enhancements. With query enhancements off, there's no guarantee if we're in the new home to use explorer as a backup so we can only prevent it since there's no way to query flint tables.